corehq/apps/userreports/tasks.py

```diff
@@ -31,6 +31,8 @@ def rebuild_indicators(indicator_config_id):
         except DataError as e:
             logging.exception('problem saving document {} to table. {}'.format(doc['_id'], e))
     adapter.engine.dispose()
+    config.built = True
+    config.save()
```

Comment:
I didn't know about `django-soil`, thanks for pointing that out.
As currently written, I check if the data source has been built every time the report is
loaded. That means I'll probably be checking the built status after the `soil.DownloadBase`
is no longer in the cache. Does soil have a good way to handle this situation?

I could still store the built status on the `DataSourceConfiguration`. Then, when the report
is loaded I would first look to get the status from the `DownloadBase`, and if it doesn't
exist, look on the `DataSourceConfiguration` itself. Seems sorta redundant though...